doc/task-hierarchy.md

```diff
@@ -0,0 +1,164 @@
+Task Type
+- Contextual Question Generation
+   - Generate questions based on given context e.g. a task to create a question based on a paragraph.
+- Answer Generation: Reading Comprehension
+- Answer Generation: Contextual Question Answering
+- Answer Generation: Open book/Domain-specific Question Answering
+- Answer Generation: Commonsense Question Answering
+- Answer Generation: Fill in the Blank
+- Answer Generation: Multiple Choice Question Answering
+- Textual Entailment
+- Incorrect Answer Generation
+- Sentence Generation
+- Paragraph Generation
+- Structured Text Generation
+   - Generate structured text in the output e.g. a task that converts questions in natural language to SQL queries
+- Topic Generation
+- Explanation Generation
+- Review Generation
+- Order Generation
+  - Given a set of elements, find their order (e.g. monotonically increasing/decreasing numbers, increasing/decreasing size in case of objects)
+- Title Generation
+- Weblink Generation
+- Text Modification
+- Classification
+- Verification 
+   - Verify if two inputs match with each other e.g. a task to verify if a sentence and its sentiment given in the input match
+- Translation
+- Hypernym Discovery
+- Parts-of-speech
+- Coreference: event coreference
+- Intent Detection
+- Claim Verification
```

Comment:
Is "Claim Verification" a subcategory of "Verification"?

Reply:
Yes, will add header